### DISCUSSION TITLE
Toggling a Modal

### DISCUSSION THREAD

**Discussion by Muratcol** [orchidsoftware/platform#1529] [Q&A] [answered]
Hello guys!

I need your help with a very simple and basic subject. I'm sorry if it's asked before. I looked but could not find it.

Can you show me an example of triggering a modal? I tried it over and over but couldn't manage it. There is a link as seen in the image, when I click this link, I want to toggle a modal in which I can set two different dates with DateTimer. Later, I want to get the dates I got from this Modal in query function as input.

This is the link:

![Screenshot (4)_LI](https://user-images.githubusercontent.com/58704132/104916296-e9fa4880-59a2-11eb-9d78-e6ba30488027.jpg)

**Comment by lemyskaman** [comment, marked as answer]
you can do that on a modal, but what are you trying to do fits better on filters functionality as selection filter layauts:

https://orchid.software/en/docs/filters/#eloquent-filter   

would help you, If you change the input text for a  date time one:

https://orchid.software/en/docs/field/#datetime

 in fact I think there is already a date range filter

**Reply by Muratcol** [reply]
Yes, However, I don't want to filter the two date ranges. I want to set two different dates with `Datetimer` as parameters, without creating crowds on the screen. So I want to set this parameter with the Modal tool.
But how can I toggle a Modal by clicking the button or link... I couldn't find it in docs?

**Reply by lemyskaman** [reply]
Just add a ModalToggle action element on the commandBar returned array of your Screen in which make()->method param, has the name of your modal name (the modal item of your screen Layout returned array).

After that Orchid screen handles the toggling 

you can verify how it is done in your App\Orchid\Screens\Examples\ExampleScreen class

**Reply by Muratcol** [reply]
Ahah, thank you so much. I always knew it's an easy thing to toggle but never find out how to do that.

**Comment by lemyskaman** [comment]
@tabuna  Perhabs the documentation at https://orchid.software/en/docs/modals/#using can be improoved making mention of  the use of the action button for modals on the command bar to allow them to work....

**Reply by tabuna** [reply]
I think it makes sense to add this.
